### DISCUSSION TITLE
How to do the "autoConnect" on demand, without setting it to "true" in the client config?

### DISCUSSION THREAD

**Discussion by Darkbound** [wevm/wagmi#2118] [Q&A]
I have a case in which I also authenticate users with JWT and all of that, if the JWT is still present, say after a refresh, I need to automatically connect the wallet of a user, but if the JWT has expired, I need to disconnect the wallet.

For the disconnect I am using:
```
await disconnectAsync();
```

And that works.

However, neither `connect()` nor `connectAsync()` from the `useConnect` hook, I can use to mimic the `autoConnect`, because both require a connector.

In this case, the user already connected previously, so all LS variables are in place.

How to "manually" invoke "autoConnect"?

I am using `Next.js`

P.S. One solution I can think of at the moment is to move the `createClient({ ... })` inside `_app.tsx` and just change the `autoConnect` on demand? Right now I am initializing the client outside of `_app.tsx`.

P.S. 2 I have also tried to use the `autoConnect` from `useClient`, but it does nothing.

**Comment by tmm** [comment]
You can call `client.autoConnect` directly. For example:

```ts
import * as React from 'react'

React.useEffect(() => {
  client.autoConnect()
}, [])
```

**Reply by Darkbound** [reply]
That is exactly what I have tried to do, but it does not work.

```ts
  useEffect(() => {
     const fetchUser = async () => {
        // ... other logic that fetches my user 

        console.log("before connect");
        await autoConnect();
    }

    fetchUser();
  }, [])
```

The log works, but the autoConnect doesnt, I dont get my provider back after a refresh.

P.S. By "it doesnt work" what I mean is that:

```ts
  const { connector, isConnected, isConnecting, isReconnecting } = useAccount();
  const { data: balance } = useBalance({ address: user.wallet });
  const { chain } = useNetwork();
```
None of these load up and give me the data

**Reply by tmm** [reply]
Can you put together a minimal reproducible example with [new.wagmi.sh](https://new.wagmi.sh)?

**Reply by Darkbound** [reply]
@tmm https://stackblitz.com/edit/new-wagmi-zngtpj?file=src%2Fcomponents%2FConnect.tsx

1. Connect Metamask, you get your account address and stuff
2. Refresh
3. You get nothing

P.S. Just fyi, I dont refresh my whole page, I am using this button https://prnt.sc/18UQ0SF9pj0H

**Reply by tmm** [reply]
This works for me. Don't destructure the the Client.

```diff
- const { autoConnect } = useClient();
+ const client = useClient();

useEffect(() => {
  const fetchUser = async () => {
    console.log('stuff...');
-   await autoConnect();
+   await client.autoConnect();
  };

  fetchUser();
}, []);
```

**Reply by Darkbound** [reply]
@tmm I confirm this worked, I was under the impression that useClient is a hook and I just take whatever I need from it, but instead it returns the instance of the class I see now, and thats why it didnt work.
